mlrun/api/api/utils.py

```diff
@@ -150,12 +156,169 @@ async def submit_run(db_session: Session, auth_info: mlrun.api.schemas.AuthInfo,
+                        if username_from_sub_path:
+                            if (
+                                username is not None
+                                and username != username_from_sub_path
+                            ):
+                                logger.warning(
+                                    "Found more than one user for volume, skipping obfuscation",
+                                    volume=volume,
+                                    volume_mounts=volume_name_to_volume_mounts[
+                                        volume_name
+                                    ],
+                                )
+                                found_more_than_one_username = True
+                                break
+                            username = username_from_sub_path
+                if found_more_than_one_username:
+                    continue
+                if not username:
+                    logger.warning(
+                        "Could not resolve username from volume mount, skipping obfuscation",
+                        volume=volume,
+                        volume_mounts=volume_name_to_volume_mounts[volume_name],
+                    )
+                    continue
+                username: str
+                secret_name = mlrun.api.crud.Secrets().store_auth_secret(
+                    mlrun.api.schemas.AuthSecretData(
+                        provider=mlrun.api.schemas.SecretProviderName.kubernetes,
+                        username=username,
+                        access_key=access_key,
+                    )
+                )
+                access_key_attribute_name = "accessKey"
+                secret_ref_attribute_name = "secretRef"
+                flex_volume_attribute_name = "flexVolume"
+                if get_item_attribute(volume, "flex_volume"):
+                    flex_volume_attribute_name = "flex_volume"
+                    access_key_attribute_name = "access_key"
+                    secret_ref_attribute_name = "secret_ref"
+
+                mlrun.runtimes.utils.set_item_attribute(
+                    get_item_attribute(
+                        get_item_attribute(volume, flex_volume_attribute_name),
+                        "options",
+                    ),
+                    access_key_attribute_name,
+                    None,
+                )
+                mlrun.runtimes.utils.set_item_attribute(
+                    get_item_attribute(volume, flex_volume_attribute_name),
+                    secret_ref_attribute_name,
+                    {"name": secret_name},
+                )
+
+
+def _obfuscate_v3io_access_key_env_var(function, auth_info: mlrun.api.schemas.AuthInfo):
+    if (
+        function.kind
+        and function.kind not in mlrun.runtimes.RuntimeKinds.local_runtimes()
+    ):
+        function: mlrun.runtimes.pod.KubeResource
+        v3io_access_key = function.get_env("V3IO_ACCESS_KEY")
```

Comment:
I think if you have already created class `FunctionEnvironmentVariables` which contains the function environment variables then `V3IO_ACCESS_KEY` should also be there.

Reply:
`V3IO_ACCESS_KEY` is not necessarily a function environment variable, it can be used anywhere where you use the v3io client package
I agree we should move it to constant, but it's not a function context constant, and in general, this specific one is all over the code so I prefer leaving it out of this PR